submariner/centralized-control-plane.md

```diff
@@ -0,0 +1,386 @@
+ enhancements to the MCS API, it would be worthwhile to experiment with different approaches,
+ such as the one defined here, before attempting to reach community consensus on MCS API changes.
+
+The design proposal attempts to achieve the above goals minimal changes to workload
+ clusters, especially with respect to data plane configuration. Ideally, the control
+ plane changes proposed will are extensions to MCS or implemented as a layer above
+ MCS and confined to the Broker cluster only.
+
+## Additional Background and Context
+
+Based on work we've done in Istio (see for example [this](https://istio.io/latest/blog/2019/isolated-clusters/)),
+ we identified two common usage models for clusters:
+
+- **Shared Infrastructure**: clusters are (mostly) identically configured (e.g., namespaces,
+  RBAC, etc.). Each team is allocated the same set of namespaces across all clusters and can
+  deploy to any cluster. The boundary between teams is the namespace. This is the model assumed
+  by MCS and Submariner.
+- **Federated**: clusters are used as the boundary between teams (e.g., they may even be
+  billed to different departments). Each team can only deploy to a subset of the clusters
+  (i.e., the ones they "own") and is granted full reign over that cluster. They are free to
+  choose whatever names they desire. Thus, the federated mode cannot assume name sameness.
+  The teams may still have a shared clusters where the "integrated application" runs, pulling
+  in services from the per team clusters.
+
+The federated usage model needs to support the following use cases (*partial list*...):
+
+- I'd like to export my internal services only to my other clusters, but still consume shared
+  services (e.g., centralized monitoring and alerting) provided by other teams' clusters.
+- I'd like to decouple the local and global names used for services and import a remote service
+  into my namespace using a different name (e.g., two teams can have an "authz" service running
+  in "frontend" namespace, but each is unique).
+- When importing a service I'd like to control its "name" and "location" (e.g., import into a
+  specific namespace with a specific name). I can't assume that the same namespaces exist
+  across all clusters and importing with the same name may conflict with an existing service.
+
+> Elaborate additional use cases here? One section per use cases, explaining motivation and
+> needed extension to the MCS objects and/or model to support the use case.
+
+## Detailed Proposal
+
+### Goals
+
+- Enable independent service name and namespace in workload clusters.
+- Enable a Service to be imported into a subset of the workload clusters only.
+- Enable centralized control over definition of Service exports and imports.
+- Enable more decoupled binding (i.e., explicit and not only based on names)
+  between existing network endpoints and the service being shared.
```

Comment:
As stated previously, all of these can be achieved by writing a different service discovery layer other than Lighthouse.

This shouldn't be part of Submariner core though, but if you feel that parts of Lighthouse can be broken into micro-services (eg DNS server component) such that they can be reused in such a solution, we can discuss that in the scope of Submariner, to make sure work is not being replicated needlessly.